app/routes.py

```diff
@@ -1,2 +1,273 @@
+            tasks = Task.query.order_by(Task.title.asc()).all() 
+        elif sort_by_url == "desc":
+            tasks = Task.query.order_by(Task.title.desc()).all() 
+        else:
+            tasks = Task.query.all()
+        # end of the new code
+    
+        tasks_response = []
+        for task in tasks:
+            if task.completed_at == None:
+                completed_at = False
+            else:
+                completed_at = True
+
+            tasks_response.append({
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": completed_at,
+            })
+        return make_response(jsonify(tasks_response), 200)
+
+
+#create a task with null completed at
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            return {"details": "Invalid data"}, 400
+            
+        if not "completed_at" in request_body or not request_body["completed_at"]:
+            completed_at = False
+        else:
+            completed_at = True
+        
+        new_task = Task(
+            title=request_body["title"],
+            description=request_body["description"], 
+            completed_at=request_body["completed_at"]
+        )
+
+        db.session.add(new_task) # "adds model to the db"
+        db.session.commit() # commits the action above
+
+        return make_response({
+            "task":{ 
+                "id": new_task.task_id,
+                "title": new_task.title,
+                "description": new_task.description,
+                "is_complete": completed_at
+            }
+        }), 201   
+
+# return one task
+@task_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"], strict_slashes = False)
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+
+    if request.method == "GET":
+        if task is None:
+            return make_response(f"Task #{task_id} not found"), 404
+        if not task.completed_at:
+            completed_at = False
+        else:
+            completed_at = task.completed_at["completed_at"]
+        return {
+            "task":{
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": completed_at
+            }
+        }, 200
+
+    # Update a task
+    elif request.method == "PUT":
+        if task is None:
+            return make_response(f"Task #{task_id} not found"), 404
+        form_data = request.get_json()
+
+        if "title" not in form_data or "description" not in form_data or "completed_at" not in form_data:
+            return {"details": "Invalid data"}, 400
+
+        if not "completed_at" in form_data or not form_data["completed_at"]:
+            is_complete = False
+        else:
+            is_complete = True            
```

Comment:
I mentioned this above, but this and the equivalent POST code could benefit from being moved to a helper function.